cmd/upgrade/app/v1alpha1/result.go

```diff
@@ -0,0 +1,258 @@
+	return stringer.Yaml("upgraderesult builder", urb)
+}
+
+// GoString implements GoStringer interface
+func (urb *UpgradeResultBuilder) GoString() string {
+	return urb.String()
+}
+
+// NewUpgradeResultBuilder returns a new UpgradeResult instance
+func NewUpgradeResultBuilder() *UpgradeResultBuilder {
+	return &UpgradeResultBuilder{
+		UpgradeResult: &UpgradeResult{},
+		errors:        []error{},
+	}
+}
+
+// WithInstanceName adds Name in UpgradeResult instance
+func (urb *UpgradeResultBuilder) WithInstanceName(name string) *UpgradeResultBuilder {
+	urb.InstanceName = name
+	return urb
+}
+
+// WithInstanceNamespace adds Namespace in UpgradeResult instance
+func (urb *UpgradeResultBuilder) WithInstanceNamespace(namespace string) *UpgradeResultBuilder {
+	urb.InstanceNamespace = namespace
+	return urb
+}
+
+// WithInstanceUID adds UID in UpgradeResult instance
+func (urb *UpgradeResultBuilder) WithInstanceUID(uid types.UID) *UpgradeResultBuilder {
+	urb.InstanceUID = uid
+	return urb
+}
+
+// WithUpgradeConfig ...adds UpgradeConfig in UpgradeResult instance
+func (urb *UpgradeResultBuilder) WithUpgradeConfig(config *upgrade.UpgradeConfig) *UpgradeResultBuilder {
+	urb.UpgradeConfig = config
+	return urb
+}
+
+// WithResourceDetails adds ResourceDetails in UpgradeResult instance
+func (urb *UpgradeResultBuilder) WithResourceDetails(resource *upgrade.ResourceDetails) *UpgradeResultBuilder {
+	urb.ResourceDetails = resource
+	return urb
+}
+
+// WithTasks adds Tasks in UpgradeResult instance
+func (urb *UpgradeResultBuilder) WithTasks(tasks []upgrade.UpgradeResultTask) *UpgradeResultBuilder {
+	urb.Tasks = tasks
+	return urb
+}
+
+// validate validates UpgradeResultBuilder instance
+func (urb *UpgradeResultBuilder) validate() error {
+	if len(urb.errors) != 0 {
+		return errors.Errorf("failed to validate: build errors were found: %v", urb.errors)
+	}
+	validationErrs := []error{}
+	if urb.InstanceName == "" {
+		validationErrs = append(validationErrs, errors.New("missing instance name"))
+	}
+	if urb.InstanceNamespace == "" {
+		validationErrs = append(validationErrs, errors.New("missing instance namespace"))
+	}
+	if urb.InstanceUID == "" {
+		validationErrs = append(validationErrs, errors.New("missing instance uid"))
+	}
+	if urb.UpgradeConfig == nil {
+		validationErrs = append(validationErrs, errors.New("missing upgrade config"))
+	}
+	if urb.ResourceDetails == nil {
+		validationErrs = append(validationErrs, errors.New("missing resource details"))
+	}
+	if len(urb.Tasks) == 0 {
+		validationErrs = append(validationErrs, errors.New("missing tasks"))
+	}
+	if len(validationErrs) != 0 {
+		urb.errors = append(urb.errors, validationErrs...)
+		return errors.Errorf("validation error(s) found: %v", validationErrs)
+	}
+	return nil
+}
+
+// Build builds a new instance of UpgradeResult with the
+// helps of UpgradeResultBuilder
+func (urb *UpgradeResultBuilder) Build() (res *upgrade.UpgradeResult, err error) {
```

Comment:
This seems like `UpgradeResultBuilder` should be named as `UpgradeResultGetOrCreateBuilder`
Similarly `Build()` may be renamed to `GetOrCreate()`

Reply:
updated